src/lib.rs

```diff
@@ -19,6 +20,14 @@ pub struct ThreadResult {
     pub result: Result<(), String>,
 }
 
+/// Wrapper struct to hold arbitrary data with a start and end timestamp
+/// denoting the time the data was captured during.
+pub struct Timestamped<T> {
+    pub start: jiff::Zoned,
+    pub end: jiff::Zoned,
```

Comment:
`jiff` suddenly became very core to the library in a way I don't fully like. Whether or not a time is "zoned" is more a property of how you want it formatted when converted to a string. It's less relevant for internal storage of a timestamp.. I think I can maybe replace most of `jiff::Zoned` with `std::time::SystemTime` and then use `jiff::Zoned::try_from(system_time)` to convert it just before serialization maybe?

Reply:
Fixed! After reading some code in `jiff` I realized that `Zoned::now()` was just a call to `SystemTime::now()` + reading of timezone anyway. It's probably just a waste to read the system timezone on every frame capture. Instead, with this solution we only sample the system clock on every frame capture. Then on serialization to JSON the oldest and newest timestamps are converted to the RFC 3339 compatible format. This feels much better.